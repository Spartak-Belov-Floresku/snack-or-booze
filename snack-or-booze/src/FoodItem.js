import { useState, useEffect } from "react";
import { Redirect, useParams, Link } from "react-router-dom";
import { Card, CardBody, CardTitle, CardText } from "reactstrap";
import SnackOrBoozeApi from "./Api";


/**
 * This component upload snack data on the UI
 */
function FoodItem() {

  const { id } = useParams();
  const [ snack, setSnack ] = useState({});

  useEffect(() => {

    const getSnack = async () => {
      try{
        const snack = await SnackOrBoozeApi.getSnack(id);
        setSnack(snack)
      }catch(err){
        setSnack(false);
      }
    }
    
    getSnack();
    
  },[]);

  if(!snack){
    return(<Redirect to={`/snacks`}/>)
  }
  
  return !snack.id?<p>Loading &hellip;</p>:(<section>
                                                <Card>
                                                  <CardBody>
                                                    <CardTitle className="font-weight-bold text-center">
                                                      {snack.name}
                                                    </CardTitle>
                                                    <CardText className="font-italic">{snack.description}</CardText>
                                                    <p>
                                                      <b>Recipe:</b> {snack.recipe}
                                                    </p>
                                                    <p>
                                                      <b>Serve:</b> {snack.serve}
                                                    </p>
                                                    <p>
                                                      <Link to={`/edit/snacks/${snack.id}`}>Edit snack</Link>
                                                    </p>
                                                    <p>
                                                      <Link to="/snacks">Snaks menu</Link>
                                                    </p>
                                                  </CardBody>
                                                </Card>
                                              </section>);
}

export default FoodItem;
