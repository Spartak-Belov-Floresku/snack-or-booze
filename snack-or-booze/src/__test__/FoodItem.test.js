import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import FoodItem from '../FoodItem';

/** render FoodItem component */
test('renders learn react link', async () => {
    render(
        <MemoryRouter initialEntries={["/snacks/nachos/" ]}>
            <Route path={"/snacks/:id"} component={FoodItem} />
        </MemoryRouter>
    );
    
    await screen.findByText("An American classic!")
});

/** Matching snapshot of FoodItem components */
test('renders learn react link', async () => {
    
    const {asFragment} =  render(
        <MemoryRouter initialEntries={["/snacks/nachos"]}>
            <Route path={"/snacks/:id"} component={FoodItem} />
        </MemoryRouter>
    );

    await screen.findByText("An American classic!")
    expect(asFragment()).toMatchSnapshot();
});