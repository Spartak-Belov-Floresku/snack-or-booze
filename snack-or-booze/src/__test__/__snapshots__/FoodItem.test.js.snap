// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <section>
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <h5
          class="font-weight-bold text-center card-title"
        >
          Nachos
        </h5>
        <p
          class="font-italic card-text"
        >
          An American classic!
        </p>
        <p>
          <b>
            Recipe:
          </b>
           Cover expensive, organic tortilla chips with Cheez Whiz.
        </p>
        <p>
          <b>
            Serve:
          </b>
           Serve in a hand-thrown ceramic bowl, garnished with canned black olives
        </p>
        <p>
          <a
            href="/edit/snacks/nachos"
          >
            Edit snack
          </a>
        </p>
        <p>
          <a
            href="/snacks"
          >
            Snaks menu
          </a>
        </p>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
